File: app/dto/workflow.py
from __future__ import annotations
from dataclasses import dataclass
from typing import Dict, Any, List, Optional
from core.workflow.model import WorkforceWorkflow, WorkerHandoff

@dataclass
class WorkflowDTO:
    id: str
    project_id: str
    title: str
    status: str
    task_ids: List[str]
    current_step: Optional[int]
    iterations: int
    created_at: str

    def to_dict(self) -> Dict[str, Any]:
        return {
            "id": self.id,
            "project_id": self.project_id,
            "title": self.title,
            "status": self.status,
            "task_ids": self.task_ids,
            "current_step": self.current_step,
            "iterations": self.iterations,
            "created_at": self.created_at
        }

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> WorkflowDTO:
        return cls(
            id=data["id"],
            project_id=data["project_id"],
            title=data["title"],
            status=data["status"],
            task_ids=data.get("task_ids", []),
            current_step=data.get("current_step"),
            iterations=data.get("iterations", 0),
            created_at=data["created_at"]
        )

    @classmethod
    def from_domain(cls, workflow: WorkforceWorkflow) -> WorkflowDTO:
        return cls(
            id=workflow.id,
            project_id=workflow.project_id,
            title=workflow.title,
            status=workflow.status.value if hasattr(workflow.status, 'value') else str(workflow.status),
            task_ids=workflow.task_ids,
            current_step=workflow.current_step,
            iterations=workflow.iterations,
            created_at=workflow.created_at
        )

@dataclass
class HandoffDTO:
    id: str
    source_worker: str
    destination_worker: str
    summary: str
    handoff_type: str
    created_at: str

    def to_dict(self) -> Dict[str, Any]:
        return {
            "id": self.id,
            "source_worker": self.source_worker,
            "destination_worker": self.destination_worker,
            "summary": self.summary,
            "handoff_type": self.handoff_type,
            "created_at": self.created_at
        }

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> HandoffDTO:
        return cls(
            id=data["id"],
            source_worker=data["source_worker"],
            destination_worker=data["destination_worker"],
            summary=data["summary"],
            handoff_type=data["handoff_type"],
            created_at=data["created_at"]
        )

    @classmethod
    def from_domain(cls, handoff: WorkerHandoff) -> HandoffDTO:
        return cls(
            id=handoff.id,
            source_worker=handoff.source_worker,
            destination_worker=handoff.destination_worker,
            summary=handoff.summary,
            handoff_type=handoff.handoff_type.value if hasattr(handoff.handoff_type, 'value') else str(handoff.handoff_type),
            created_at=handoff.created_at
        )

File: client/lib/models/workflow.dart
class WorkflowInfo {
  final String id;
  final String projectId;
  final String title;
  final String status;
  final List<String> taskIds;
  final int currentStep;
  final int iterations;
  final String createdAt;

  const WorkflowInfo({
    required this.id,
    required this.projectId,
    required this.title,
    this.status = 'PENDING',
    this.taskIds = const [],
    this.currentStep = 0,
    this.iterations = 0,
    required this.createdAt,
  });

  factory WorkflowInfo.fromJson(Map<String, dynamic> json) {
    return WorkflowInfo(
      id: json['id'] as String? ?? '',
      projectId: json['project_id'] as String? ?? '',
      title: json['title'] as String? ?? '',
      status: json['status'] as String? ?? 'PENDING',
      taskIds: (json['task_ids'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? const [],
      currentStep: json['current_step'] as int? ?? 0,
      iterations: json['iterations'] as int? ?? 0,
      createdAt: json['created_at'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'project_id': projectId,
    'title': title,
    'status': status,
    'task_ids': taskIds,
    'current_step': currentStep,
    'iterations': iterations,
    'created_at': createdAt,
  };
}

class HandoffInfo {
  final String id;
  final String sourceWorker;
  final String? destinationWorker;
  final String summary;
  final String handoffType;
  final String createdAt;

  const HandoffInfo({
    required this.id,
    required this.sourceWorker,
    this.destinationWorker,
    required this.summary,
    this.handoffType = 'GENERAL',
    required this.createdAt,
  });

  factory HandoffInfo.fromJson(Map<String, dynamic> json) {
    return HandoffInfo(
      id: json['id'] as String? ?? '',
      sourceWorker: json['source_worker'] as String? ?? '',
      destinationWorker: json['destination_worker'] as String?,
      summary: json['summary'] as String? ?? '',
      handoffType: json['handoff_type'] as String? ?? 'GENERAL',
      createdAt: json['created_at'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'source_worker': sourceWorker,
    'destination_worker': destinationWorker,
    'summary': summary,
    'handoff_type': handoffType,
    'created_at': createdAt,
  };
}
